java/client/src/main/java/glide/api/models/commands/FT/FTProfileOptions.java

```diff
@@ -0,0 +1,175 @@
+/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands.FT;
+
+import static glide.api.models.GlideString.gs;
+import static glide.utils.ArrayTransformUtils.concatenateArrays;
+
+import glide.api.commands.servermodules.FT;
+import glide.api.models.GlideString;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+import lombok.NonNull;
+
+/** Mandatory parameters for {@link FT#profile} command. */
+public class FTProfileOptions {
+    private final QueryType queryType;
+    private final boolean limited;
+    private final GlideString[] query;
+
+    /** Query type being profiled. */
+    public enum QueryType {
+        SEARCH,
+        AGGREGATE
+    }
+
+    /**
+     * Profile a query given as an array of module command line arguments.
+     *
+     * @param queryType The query type.
+     * @param commandLine Command arguments (not including index name).
+     */
+    public FTProfileOptions(@NonNull QueryType queryType, @NonNull GlideString[] commandLine) {
+        this(queryType, commandLine, false);
+    }
+
+    /**
+     * Profile a query given as an array of module command line arguments.
+     *
+     * @param queryType The query type.
+     * @param commandLine Command arguments (not including index name).
+     */
+    public FTProfileOptions(@NonNull QueryType queryType, @NonNull String[] commandLine) {
+        this(queryType, commandLine, false);
+    }
+
+    /**
+     * Profile a query given as an array of module command line arguments.
+     *
+     * @param queryType The query type.
+     * @param commandLine Command arguments (not including index name).
+     * @param limited Either provide a full verbose output or some brief version (limited).
+     */
+    public FTProfileOptions(
+            @NonNull QueryType queryType, @NonNull GlideString[] commandLine, boolean limited) {
+        this.queryType = queryType;
+        this.query = commandLine;
```

Comment:
Why is this called commandLine? It should be query.

Reply:
Actually it is a complete command line for aggregate/search except first 2 args